src/compiler/crystal/syntax/lexer.cr

```diff
@@ -1548,7 +1544,7 @@ module Crystal
         raw_number_string = raw_number_string.delete('_') if has_underscores
         @token.value = raw_number_string
       else
-        base10_number_string = raw_number_string.to_u64?(base: base, underscore: true).try &.to_s
+        base10_number_string = raw_number_string.to_u128?(base: base, underscore: true).try &.to_s
```

Comment:
I think it would be better and even more performant (so, regardless of Windows support) to keep `to_u64` if the number still fits into that, otherwise try `to_u128`.

Reply:
Maybe we don't need to be too smart about it, just `(.to_u64?(...) || .to_u128?(...)).try ...`

Reply:
I did that in the previous PR, and only changed to make the code simpler. I can change it (previously I checked if `bits_per_digit ** number_size > 64`, but that will fall back to i128 with very big octal literals because 3 % 64 != 0.

Reply:
For big octal literals I had a proper check that you can get inspiration from.
https://github.com/crystal-lang/crystal/commit/6eb0bb1354ac34e701c634c1b006f3a07744fb65#diff-f038607b299e04ef78ff1b13b738bc152173cf18d156b412ce324941eb7426ddR1827

But I don't think that's what should be done.
Again, I really think this is the way to go:
`(.to_u64?(...) || .to_u128?(...)).try ...`

Reply:
Is there even a significant performance difference?

Reply:
@straight-shoota If we ever have bugs with int128 support, we really don't want that to affect absolutely everything in any case.

Reply:
> Is there even a significant performance difference?

Just tested, it's basically 1µs when only using u128 and 800ns when first using u64 (string `0xffffffffffffffff`, improvement 24%), which is also the biggest improvement I found (could be more severe on 32-bit systems). But anyways, I don't think that the performance is that important in the lexer when the compiler spends most of it's time in LLVM anyways.

Reply:
@straight-shoota You can see in the earlier CI job that bugs in int128 support are not hypothetical, so this PR wouldn't immediately cause a regression on Windows and maybe other platforms that we're not aware of

Reply:
Okay, let's do this. But please leave a comment about the reasoning behind parsing first i64 and falling back to i128.